src/components/MultiHopTrade/components/SharedTradeInput/SharedRecipientAddress.tsx

```diff
@@ -0,0 +1,371 @@
+import { Row } from 'components/Row/Row'
+import { RawText, Text } from 'components/Text'
+import type { TextPropTypes } from 'components/Text/Text'
+import { useAccountsFetchQuery } from 'context/AppProvider/hooks/useAccountsFetchQuery'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { useIsSnapInstalled } from 'hooks/useIsSnapInstalled/useIsSnapInstalled'
+import { useModal } from 'hooks/useModal/useModal'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import {
+  useWalletSupportsChain,
+  useWalletSupportsChainAtRuntime,
+} from 'hooks/useWalletSupportsChain/useWalletSupportsChain'
+import { parseAddressInputWithChainId } from 'lib/address/address'
+import { middleEllipsis } from 'lib/utils'
+import { selectAccountIdsByAssetId } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+const editIcon = <EditIcon />
+const checkIcon = <CheckIcon />
+const closeIcon = <CloseIcon />
+
+type ManualRecipientAddressLabelsProps = {
+  buyAsset: Asset
+  manualAddressEntryDescription: string | undefined
+}
+
+const ManualRecipientAddressLabels = ({
+  buyAsset,
+  manualAddressEntryDescription,
+}: ManualRecipientAddressLabelsProps) => {
+  const translate = useTranslate()
+  const wallet = useWallet().state.wallet
+  const { isSnapInstalled } = useIsSnapInstalled()
+  const { open: openSnapsModal } = useModal('snaps')
+  const { open: openManageAccountsModal } = useModal('manageAccounts')
+
+  const walletSupportsBuyAssetChainAtRuntime = useWalletSupportsChainAtRuntime(
+    buyAsset.chainId,
+    wallet,
+  )
+
+  const buyAssetChainName = useMemo(() => {
+    const chainAdapterManager = getChainAdapterManager()
+    return chainAdapterManager.get(buyAsset.chainId)?.getDisplayName()
+  }, [buyAsset])
+
+  const isMetaMaskWalletWithoutSnap = useMemo(() => {
+    return !isSnapInstalled && wallet && isMetaMask(wallet)
+  }, [isSnapInstalled, wallet])
+
+  // We're enabling the snap, so no versioning concerns here
+  const handleEnableShapeShiftSnap = useCallback(() => openSnapsModal({}), [openSnapsModal])
+  const handleAddAccount = useCallback(() => openManageAccountsModal({}), [openManageAccountsModal])
+
+  return (
+    <>
+      <FormLabel color='yellow.400'>
+        {manualAddressEntryDescription ??
+          translate('trade.receiveAddressDescription', { chainName: buyAssetChainName })}
+        {isMetaMaskWalletWithoutSnap && (
+          <Link textDecor='underline' ml={1} onClick={handleEnableShapeShiftSnap}>
+            {translate('trade.or')}
+            &nbsp;{translate('trade.enableMetaMaskSnap')}
+          </Link>
+        )}
+        {walletSupportsBuyAssetChainAtRuntime && (
+          <>
+            &nbsp;
+            {translate('common.or')}
+            <Link textDecor='underline' ml={1} onClick={handleAddAccount}>
+              {translate('trade.connectChain', { chainName: buyAssetChainName })}
+            </Link>
+          </>
+        )}
```

Comment:
Here's where we tell users to connect the wallet to the chain in question, if its an option. Previously this was only working for ledger.

Reply:
🐐